### ISSUE TITLE
RenderTexture has a problem with png which has alpha < 255

### ISSUE THREAD

**Issue by cx-github** [cocos2d/cocos2d-x#19956]
- cocos2d-x version:3.17.1  or all versions
- devices test on: mac
- developing environments
   - NDK version:
   - Xcode version: 
   - VS version:
   - browser type and version:

- desc
    - My English is not very good 
    - when i use rendertexture object with png ,i find png with alpha that less than 255 , result  not right 
   - For better testing ，Use layerColor instead，but It doesn't affect result 
    - look blow test
    

```


require "cocos.init"

local size = {width = 480,height =320}
local bg = cc.LayerColor:create(cc.c4b(255,255,255, 255), size.width, size.height)

local scene = cc.Scene:create()
cc.Director:getInstance():runWithScene(scene)

scene:addChild(bg)
local renderNode = cc.RenderTexture:create(size.width,size.height,cc.TEXTURE2_D_PIXEL_FORMAT_RGB_A8888, gl.DEPTH24_STENCIL8_OES)

scene:addChild(renderNode)
renderNode:beginWithClear(0,0,0,0)


local layer1 = cc.LayerColor:create(cc.c4b(255, 0, 255, 100), 50,50)
layer1:setPosition(50,50)
layer1:visit()
local layer11 = cc.LayerColor:create(cc.c4b(255, 0, 0, 100), 50,50)
layer11:setPosition(50,70)
layer11:visit()
renderNode:endToLua()

local sprite = renderNode:getSprite()
sprite:setAnchorPoint(cc.p(0,0))
sprite:setPosition(0,0)


local layer2 = cc.LayerColor:create(cc.c4b(255, 0, 255, 100), 50,50)
scene:addChild(layer2)
layer2:setPosition(100,50)
local layer21 = cc.LayerColor:create(cc.c4b(255, 0, 0, 100), 50,50)
scene:addChild(layer21)
layer21:setPosition(100,70)

```

![image](https://user-images.githubusercontent.com/6309189/61573133-ecdd5a00-aadb-11e9-9180-e59d937b4df4.png)

**Comment by cx-github** [comment]
@minggo

**Comment by rh101** [comment]
Are you saving the RenderTexture output image to a file with `RenderTexture::saveToFile()`?  If that is the case, and the PNG file is the problem, then there is a possible fix that has already been merged in: #19782 

With that merge you can use `RenderTexture::saveToFileAsNonPMA()`

**Comment by cx-github** [comment]
> Are you saving the RenderTexture output image to a file with `RenderTexture::saveToFile()`? If that is the case, and the PNG file is the problem, then there is a possible fix that has already been merged in: #19782
> 
> With that merge you can use `RenderTexture::saveToFileAsNonPMA()`

it is not same question
